src/context.rs

```diff
@@ -315,32 +315,26 @@ pub unsafe fn dc_get_userdata(context: &mut Context) -> *mut libc::c_void {
     context.userdata as *mut _
 }
 
-pub unsafe fn dc_open(
-    context: &Context,
-    dbfile: *const libc::c_char,
-    blobdir: *const libc::c_char,
-) -> libc::c_int {
-    let mut success = 0;
+pub unsafe fn dc_open(context: &Context, dbfile: &str, blobdir: Option<&str>) -> bool {
```

Comment:
Hmm, `dbfile` and `blobdir` should have been `Path` IMHO

Reply:
Well, not so rusty then ;) At least I have removed some raw pointers here.